server/src/controllers/course.ts

```diff
@@ -405,31 +420,13 @@ export async function getCourse(req: Request, res: Response): Promise<Response>
       }
     });
 
-    return res.status(200).send({
+    res.status(200).send({
       success: true,
       course: courseData
     });
+    return;
   } catch (error: unknown) {
-    console.log(error);
-
-    if (error instanceof yup.ValidationError) {
-      return res.status(400).send({
-        success: false,
-        error: error.errors
-      });
-    }
-
-    if (error instanceof Error && error?.message.startsWith('course with an id')) {
-      return res.status(404).send({
-        success: false,
-        error: error.message
-      });
-    }
-
-    return res.status(500).send({
-      success: false,
-      error: 'Internal Server Error'
-    });
+    next(error);
```

Comment:
Would it be reasonably possible to let this error propagate without catching it here?

Reply:
According to express [docs](https://expressjs.com/en/guide/error-handling.html), with async functions we have to use try...catch and pass them to the next. Seems like that with express 5, there is no need for try...catch as express will automatically catch and handle errors. Too bad its still in beta 🥲

Reply:
Ah, that's very unfortunate. Could we write a higher-order function that automatically converts an asynchronous middleware to one that passes rejections to Express through the `next` function, to avoid this boilerplate and chances of missing errors? Let's say,

```typescript
function handleErrors(func: (Request, Response, NextFunction) => Promise<void>): (Request, Response, NextFunction) => Promise<void> {
    return async (req: Request, res: Response, next: NextFunction) => {
        try {
            return func(req, res, next);
        } catch (err: unknown) {
            next(err);
        }
    };
}
```

(Pardon my TypeScript, I wrote this without using a real code editor).

Reply:
Added `handleErrors` function to the routes 024fd9c1d56361742eef5483c0fb0fd428934094.

I had problems with missing properties (+80) and could not find any quick solutions for the problem except using `any` and disabling linter for those. I think maybe we need to create some kind of custom type that has all the missing properties?